### DISCUSSION TITLE
how to use standalone tools from eclipse cdt ?

### DISCUSSION THREAD

**Discussion by mahatt** [eclipse-cdt/cdt#403] [General]
Hello,
I have been using  eclipse for long time and i saw others getting use to, mastering use of eclipse. Out of most discussions we had about IDE, most of use believed about features that can be used from command line or standalone ( like standalone debugger). I though it was  intuitive to have such process of development, i was wondering if there is any way to do it?  if not, why was it not preferred choice?

If i were to create it from self-written "driver" code, what challenges (technical or else ) would i face?
 
Thanks in advance.

**Comment by jonahgraham** [comment]
Sorry that I don't fully follow your question. It may be because my interpretation of your question is much too broad and out of scope to really provide you an answer.

Perhaps you can ask a narrower question and I can try to answer it?

**Reply by mahatt** [reply]
Yes I can give concrete use case. Eclipse CDT support perf tool. It includes actual "perf" tool and "perf-gui". Can i just run  "perf-gui" without starting eclipse,creating project etc. steps?  i like standalone debugger part, it made life easy, hence the expectation.

If this is not supported yet,  is it possible to write simple "driver" code and shell script to do it?
Thank you

**Reply by jonahgraham** [reply]
You can certainly run third-party tools like perf outside of Eclipse CDT. I don't know what perf-gui is, but based on the question my guess is it is the UI for perf provided by the [Eclipse Linuxtools project](https://wiki.eclipse.org/Linux_Tools_Project/PERF/User_Guide)? 

If my above assumption is correct, you want to know what it would take to convert that feature of Linuxtools into its own standalone Eclipse installation?

That is certainly out of scope of what I can answer - you may get better answers by speaking to [Linux tools project](https://github.com/eclipse-linuxtools/org.eclipse.linuxtools) (CDT is a distinct project, but both are Eclipse projects and some of the same people are active on both).

To help you answer your question yourself you could look at what code was needed to enable the CDT Debug functionality as a standalone product. Look at the `*application*` and `*standalone*` projects in https://github.com/eclipse-cdt/cdt/tree/main/debug
